src/Serialization/Groups.jl

```diff
@@ -276,6 +276,18 @@ function load_object(s::DeserializerState, ::Type{FinGenAbGroupElem}, G::FinGenA
   return G(vec(load_object(s, Matrix{ZZRingElem})))
 end
 
+# homomorphisms
+@register_serialization_type FinGenAbGroupHom uses_id
+
+function save_object(s::SerializerState, h::FinGenAbGroupHom)
+  save_object(s, (rels(domain(h)), rels(codomain(h)), matrix(h)))
```

Comment:
```suggestion
  save_object(s, (domain(h), codomain(h), matrix(h)))
```
FinGenAbGroups should be able to serialize themselves, and that way uuids will be checked by the serialization framework as well, thus keeping identities

but I think it would be even better to name the different parts like
```suggestion
  save_data_dict(s) do
    save_object(s, domain(h), :domain)
    save_object(s, codomain(h), :codomain)
    save_object(s, matrix(h), :matrix)
  end
```
as that is more readable and future-proof

Reply:
You quick for me - this is a restructure of https://github.com/oscar-system/Oscar.jl/pull/5064 and I was not yet done collecting the changes when you reviewed.

Reply:
Please look again.